src/with-component.js

```diff
@@ -6,9 +6,8 @@ import { withRender } from './with-render';
 import { withUnique } from './with-unique';
 
 export const withComponent = (Base = HTMLElement) => class extends withUnique(withRender(withProps(Base))) {
-  propsChangedCallback (next, prev) {
-    super.propsChangedCallback(next, prev);
-    this.rendererCallback(render);
+  rendererCallback (shadowRoot, renderCallback) {
+    render(renderCallback(), shadowRoot);
```

Comment:
This makes it so they can decide when to "render", but also not have to worry about the function signature, making it consistent across renderers.